### PR TITLE
Testcafe E2E test

### PR DESCRIPTION
**Pull request by macobo**
This PR adds a new E2E test, which:

1. Can be run on multiple browsers (including IE11) using browserstack
2. Is run on every PR via github actions
3. Interacts with a webpage and sends events to (running) instance of posthog and checks whether the events can be later queried.

This depends on https://github.com/PostHog/posthog/pull/3173 - once that's merged, a line from this needs to be changed.

Tests are currently failing because of a legit IE11 bug this helped catch: https://github.com/PostHog/posthog/pull/3178

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: testcafe/e2e.spec.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                console.log(`Browser ${level}:`, line)
+            })
+        })
+
+        console.debug('Requests to posthog:', JSON.stringify(captureLogger.requests, null, 2))
+    })
+
+test('Captured events are accessible via /api/event', async (t) => {
+    await t
+        .click('[data-cy-custom-event-button]')
```
Comment (EDsCODE) [inline testcafe/e2e.spec.js:23]: Test and helpers look good. One thing I don't understand is where this attribute exists and how it's being clicked

**Review by EDsCODE — APPROVED** [review]
lgtm. Just have a few clarifying questions

File: .github/workflows/testcafe.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        run: yarn; yarn build-array; mkdir -p posthog/frontend/dist; cp dist/array.js posthog/frontend/dist; python posthog/manage.py collectstatic --noinput
+
+      - name: Start server
+        run: cd posthog; python manage.py setup_dev --no-data; bin/docker &
+
+      - name: Run chrome test
+        run: npx testcafe "browserstack:chrome" testcafe/*.spec.js
+
+      - name: Run ie test
+        run: npx testcafe "browserstack:ie" testcafe/*.spec.js
```
[the remark this answers is not in the document]
Comment (macobo) [reply .github/workflows/testcafe.yml:78]: It's trivial - shall we? :)

File: testcafe/e2e.spec.js (hunk 1 above)
Comment (macobo) [reply testcafe/e2e.spec.js:23]: It comes from `playground/cypress/index.html` which was previously used by cypress tests as well.

File: .github/workflows/testcafe.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (macobo) [reply .github/workflows/testcafe.yml:78]: Done!

**Pull request merged**